lib/ansible/modules/packaging/os/pkgng.py

```diff
@@ -189,17 +194,35 @@ def install_packages(module, pkgng_path, packages, cached, pkgsite, dir_arg):
             module.fail_json(msg="Could not update catalogue")
 
     for package in packages:
-        if query_package(module, pkgng_path, package, dir_arg):
+        already_installed = query_package(module, pkgng_path, package, dir_arg)
+        if already_installed and state != "latest":
             continue
 
         if not module.check_mode:
-            if old_pkgng:
-                rc, out, err = module.run_command("%s %s %s install -g -U -y %s" % (batch_var, pkgsite, pkgng_path, package))
+            if already_installed:
+                # Check to see if a package upgrade is available.
+                # rc = 0, no updates available
+                # rc = 1, updates available
+                if old_pkgng:
+                    rc, out, err = module.run_command("%s %s %s upgrade -n -g -U -y %s" % (batch_var, pkgsite, pkgng_path, package))
```

Comment:
not a show stopper, but the code might be clearer if we just add the '-n' and '%s'/dir_arg in the `if/else` and just execute `run_command` at the end